src/transformers/modeling_utils.py

```diff
@@ -1674,7 +1677,12 @@ def save_pretrained(
                     del state_dict[ignore_key]
 
         # Shard the model if it is too big.
-        weights_name = SAFE_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
+        variant_suffix = f".{variant}" if variant is not None else ""
+        if safe_serialization:
+            weights_name = SAFE_WEIGHTS_NAME + variant_suffix
+        else:
+            weights_name = WEIGHTS_NAME + variant_suffix
```

Comment:
nit: maybe like the following for consistency with the previous style and https://github.com/huggingface/transformers/pull/21332/files#diff-6b72b98c4c2dcfc6cc606843917733f5d858374fbc22a735ff483bbc0c1e63eaR1714-R1715

```suggestion
        weights_name = SAFE_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
        weights_name = weights_name + variant_suffix
```

Reply:
This is moot with the new name suggested. I'd still keep the code like it was and add the variant after:
```py
weights_name = SAFE_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
if variant is not None:
    splits = weights_name.split(".")
    splits = splits[:-1] + [variant] + splits[-1]
    weights_name = ".".join(splits)
```